src/API/FileController.ts

```diff
@@ -46,67 +45,69 @@ export interface Arguments {
+}
 
-  private log = (logType: LogTypes) => (label: string) => async <T extends any>(value?: T ) => {
-    this.logger[logType]({ value }, label)
-    return value
+export const FileController = ({
+  dependencies: {
+    fileDao,
+    logger,
+  },
+  configuration: {
+    ipfsUrl,
+    ipfsArchiveUrlPrefix,
+  },
+}: Arguments) => {
+
+  const log = (logType: LogTypes) => (label: string) => <T>(v?: T ) => {
+    logger[logType]({ value: v }, label)
+    return v
   }
 
-  private uploadStream = (stream: fs.ReadStream) => {
+  const uploadStream = (stream: fs.ReadStream) => {
     const formData = new FormData()
     formData.append('file', stream)
-    return fetch(`${this.ipfsUrl}/api/v0/add`, {
+    return fetch(`${ipfsUrl}/api/v0/add`, {
       method: 'post',
       body: formData,
       timeout: minutesToMiliseconds(10),
     })
   }
 
-  private getResponseJson = (x: Response) => x.json()
+  const getResponseJson = (x: Response) => x.json()
 
-  private addFileToIPFS = (fileStream: fs.ReadStream) => this.uploadStream(fileStream)
+  const addFileToIPFS = (fileStream: fs.ReadStream) => uploadStream(fileStream)
 
-  private storeIPFSHash = async (response: FileResponseJson) => {
+  const storeIPFSHash = async (response: FileResponseJson) => {
     const { hash } = response
-    await this.fileDao.addEntry({ hash })
+    await fileDao.addEntry({ hash })
     return response
   }
 
-  private convertResponse = (ipfsFileResponse: IPFSFileResponseJson): FileResponseJson =>
-    convertJson(this.ipfsArchiveUrlPrefix)(ipfsFileResponse)
-
-  private handleFile = asyncPipe(
-    this.log(LogTypes.trace)('Adding file to ipfs'),
-    this.addFileToIPFS,
-    this.getResponseJson,
-    this.convertResponse,
-    this.log(LogTypes.trace)('Added file to ipfs, now saving the hash to the database'),
-    this.storeIPFSHash,
-    this.log(LogTypes.trace)('Saved the hash to the database'),
+  const convertResponse = (ipfsFileResponse: IPFSFileResponseJson): FileResponseJson =>
+    convertJson(ipfsArchiveUrlPrefix)(ipfsFileResponse)
+
+  type handleFile = (stream: fs.ReadStream) => Promise<{ hash: string }>
+
+  const handleFile: handleFile = asyncPipe(
+    log(LogTypes.trace)('Adding file to ipfs'),
+    addFileToIPFS,
+    getResponseJson,
+    convertResponse,
+    log(LogTypes.trace)('Added file to ipfs, now saving the hash to the database'),
+    storeIPFSHash,
+    log(LogTypes.trace)('Saved the hash to the database'),
   )
 
-  public addFiles = asyncPipe(
-    this.log(LogTypes.trace)('Adding files'),
-    map(this.handleFile),
-    Promise.all.bind(Promise), // bind required else TypeError "Promise.all called on non-object"
-    this.log(LogTypes.trace)('Added files'),
+  const addFiles: FileController['addFiles'] = asyncPipe(
+    log(LogTypes.trace)('Adding files'),
+    (xs) => xs.map(handleFile),
+    (xs) => Promise.all(xs),
+    (xs) => log(LogTypes.trace)('Added files')(xs),
```

Comment:
For some really weird reason, I have to give this a function like this... It doesn't make any sense. It might have more to do with the asyncPipe weirdness I've been seeing. Il experiment with `pipeP`